Frankenstein Station 13: Introducing dissections and alien organs to xenobiology

EmeraldCandy opened the pull request:
## What Does This PR Do

The PR page should be primarily for issue reporting. Please direct general chatter to: 
https://discord.com/channels/145533722026967040/1362583282692395058

for a list of all organ abilities, functions, and progress, refer to the excel document here:
https://docs.google.com/spreadsheets/d/19WGbyGu1gzqhWUTsYu4SBNqetkCBr8vkhoc-zyoaXUY/edit?usp=sharing

Xenobiology has lacked content for a long time. Slimes are neat, but it gets pretty boring and many times is abandoned halfway through the shift. Now, xenobiologists have a handfull of new things that they can pursue over the course of the shift: Xeno organs! (not xenomorph. Okay well, not specifically xenomorph). This PR introduces the ability to dissect organs out of alien life, and allows them to be revitalized via slime therapy to turn into useful organs. These organs can then be implanted into others, or simply processed in the new Organ Analyzer machine to turn into research credits and potentially high-tier research disks!

**Dissecting**
Dissecting is a fairly straightforward process. All dissections begin with the dissection manager, though this can only be started on creatures that contain a xeno organ at all. This then follows normal surgical steps: Scalpel, hemostat, retractors, and finishing by pulling out the organ with the dissection manager. While there is a standard dissection procedure, not all creatures are created equal! Some creatures have different steps to open their surgical site. Ash drakes, for example, will require a bone saw to cut through their mighty scales, then removing said scales via a hemostat before you can reach its squishy flesh. Blobbernaughts, on the other hand, require retracting the incision back open every other step because THEIR GOOPY BODY KEEPS CLOSING THE DAMNED INCISION SITE! 

**The Organs**
After dissection, the scientist will be rewarded with an unidentified organ. This organ, while currently useless, can have its potential unlocked by revitalizing it via slime therapy in xenobiology (listed below). Each creature will have a list of possible organs available to them, with each most creatures having at least one unique organ available to them. Seeing as only one can be pulled from each creature, multiple corpses may be required to obtain a desired organ. Each organ will have a randomized icon, so good luck telling them apart!

Each organ also has one of three quality levels: Damaged, normal, and pristine. This is nfluenced by two factors: The dissection manager level used, and any smith tool bit modifications attached to the device. Tools with a better failure rate chance will net, on average, higher quality organs! Most of the time this will only marginally make that organ function better, however occasionally a pristine (or even damaged) quality will have unique effects. 

**Slime Revitalization Therapy**
Unidentified organs will be unusable in their current state, being pulled from a corpse and all. These organs will need to be inserted into a slime in order to revive it. However, not any ol slime will do. Slimes must be befriended before they will accept foreign object inside of them. First one needs to  discipline a slime, which can be done a few ways. A slime that tries to eat someone out of hunger, then shoved off will have a chance to discipline it. A slime hit by an object of sufficient force will have a low chance a disciplining it, but beware of retaliation! Spray bottles filled with water are also a great way to discipline slimes, however this only works if they are chasing someone or are trying to eat them. Extinguishers wont work for this, itll just make them mad. Especially trained slimes! The most consistent way to discipline however, is to feed a slime 5 sheets of plasma, which will make it very happy! 

After a slime is sufficiently disciplined, it will require an additional 5 sheets of plasma to cement its training. A trained slime will be able to take xeno organs into itself and revitalize them. A trained slime will be noticeable by the ":3" big smile on its face! happy slime! After a while itll eject the revitalized organ, ready to be implanted or analyzed! Additionally, trained slimes are less likely to become aggressive when abused or hungry. 

Would you believe most this slime behavior was already in the game before this PR?

**Organ Analyzer**
The organ analyzer is fairly straight forward. A newly revitalized is slotted into the machine, after which it can be activated with an empty hand to start the process. When its done after a few seconds, the organ will be destroyed, but will grant the science department credits based off of what organ was destroyed, and potentially generate a high-level research disk based off of what kind of organ it was. Its a give-and-take!

## Why It's Good For The Game

Xenobio and science in general has needed content. Xeno bio needs more reasons to get out into the station/world and do more and have more reasons to talk to people. From shift start, xeno can get every color slime in 30-40 minutes and then they have nothing else to do. (at least I can). Slime ranching is alright, its pretty cathartic, but theres not much at the end of the road to achieve. Now, we can make HORRIBLE MONSTROSITIES out of crew!

As a side effect from this PR several changes have occured:
all megafauna have been granted a corpse sprite, so that there is a body to extract the organs from. 
All megafauna corpses are un-pullable. Mr science man is going to have to get his hands dirty down there.
Organs datum had a bit of a refactor to account for all the new processes, thus abductor organs were affected (but should be unchanged from their original design).

## Images of changes

![image](https://github.com/user-attachments/assets/4d902af3-8ad4-40c2-b57b-4e1b188bf2a8)
![image](https://github.com/user-attachments/assets/e56e8285-c024-443e-b0b7-86f6b629202e)
![image](https://github.com/user-attachments/assets/c36a78fc-a611-411c-8911-434f542db2cf)
![image](https://github.com/user-attachments/assets/b350b971-3464-4e5d-a6df-9542806b60b7)
![image](https://github.com/user-attachments/assets/39611ccb-6cdd-4d59-884a-7397099b8675)


## Todo:

- [ ] Implement organ quality
- [ ] give megafauna corpses proper
- [ ] implement tool bits to dissection managers
- [ ] code every. single. organ AND their quality effects
- [ ] assign every eligible creature their potential organ list
- [ ] assign every eligible creature their surgery list (most are default)

<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  <!-- Replace the box with [x] to mark as complete. -->
  <!-- Ensure there are no spaces between the x and the square brackets [] else this will not work properly. -->

![image](https://github.com/user-attachments/assets/2bf52580-2fbf-4c22-b066-10b90fe3aefc)
![image](https://github.com/user-attachments/assets/0d854882-1a80-4576-bbed-5d44a6732985)
![image](https://github.com/user-attachments/assets/289578c1-f1d8-40da-ba0e-5d2203bafc85)


<hr>

## Changelog

:cl:
add: Added xeno organs, which can be revitalized for unique effects
add: Added the ability to dissect xeno organs out of alien creatures
add: added the organ analyzer machine
add: added the dissection manager surgery tool
add: Added slime behavior for training and revitalization therapy for xeno organs
tweak: Tweaked a few items into the new xeno organ system (Xenomorph organs, colossus vocal coords)
soundadd: Added a sound for the organ analyzer
imageadd: Added megafauna corpse sprites
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

ArcticValor commented:
Oh no, we're stealing xeno livers for NT now......

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+///allows humanoids to eat arbitrary items
+/datum/component/special_tastes
+	///associated list of item to reagents and tastes
```
```suggestion
	/// associated list of item to reagents and tastes
```

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+///allows humanoids to eat arbitrary items
+/datum/component/special_tastes
+	///associated list of item to reagents and tastes
+	var/list/obj/item/edible_items = list() //Typepath = (list(reagents = list(reagents), tastes = list(tastes)); Ex: list(/obj/item/stack/sheet/mineral/silver = list("reagents" = list("nutriment" = 5, "vitamin" = 1), "tastes" = list("metal and blood" = 1)))
```
```suggestion
	var/list/obj/item/edible_items = list() // Typepath = (list(reagents = list(reagents), tastes = list(tastes)); Ex: list(/obj/item/stack/sheet/mineral/silver = list("reagents" = list("nutriment" = 5, "vitamin" = 1), "tastes" = list("metal and blood" = 1)))
```

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+	for(var/obj/item/current_item as anything in edible_items)
+		if(!islist(edible_items[current_item])) //if not a list of reagents
```
```suggestion
		if(!islist(edible_items[current_item])) // if not a list of reagents
```

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+	src.edible_items = edible_items
+	RegisterSignal(parent, COMSIG_ATTACK_BY, PROC_REF(attempt_ingest))
+
+///check if item is in the list of potential food
```
```suggestion
/// check if item is in the list of potential food
```

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+///check if item is in the list of potential food
+/datum/component/special_tastes/proc/attempt_ingest(source, obj/item/attacking_item, mob/living/carbon/human/attacker)
+	SIGNAL_HANDLER
+	if(parent != attacker) //only if they attack themselves with it
```
```suggestion
	if(parent != attacker) // only if they attack themselves with it
```

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+	attempt_to_eat(attacker, attacking_item, similar_to)
+	return COMPONENT_SKIP_AFTERATTACK
+
+///handle the
```
Handle the what?

EmeraldCandy replied:
fixed

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+	var/fullness = user.nutrition + 10
+	for(var/datum/reagent/consumable/chem in food.reagents.reagent_list)
+		fullness += chem.nutriment_factor * chem.volume / (chem.metabolization_rate * user.metabolism_efficiency)
+	if(!(fullness > (600 * (1 + user.overeatduration / 2000)))) //need to work around the sleep()'s in eat()
```
```suggestion
	if(!(fullness > (600 * (1 + user.overeatduration / 2000)))) // need to work around the sleep()'s in eat()
```

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+		user.consume(food)
+		food.On_Consume(user, user)
+		qdel(attacking_item)
+		return //couldnt eat it, undo
```
```suggestion
		return // couldnt eat it, undo
```

PollardTheDragon commented on code/datums/components/special_tastes.dm:
```diff
@@ -0,0 +1,75 @@
+	user.put_in_active_hand(attacking_item)
+	qdel(food)
+
+///turn the item into food, can only be eaten in 1 bite to prevent
```
Prevent what?

EmeraldCandy replied:
fixed

PollardTheDragon commented on code/datums/components/surgery_initiator.dm:
```diff
@@ -80,6 +80,12 @@
+		to_chat(user, "<span class = 'warning'>You cannot begin surgery on a patient with an open dissection site!</span>")
+		return
+	if(isalien(L)) // god why do xeno organs have to be fundamental to how they live. This is easier.
+		to_chat(user, "<span class = 'warning'>You're going to do a proper dissection to get anything useful from this.</span>")
```
```suggestion
		to_chat(user, "<span class = 'warning'>You're going to have to do a proper dissection to get anything useful from this.</span>")
```

PollardTheDragon commented on code/game/jobs/access.dm:
```diff
@@ -107,7 +107,8 @@
 				ACCESS_INTERNAL_AFFAIRS, ACCESS_LIBRARY, ACCESS_MAGISTRATE, ACCESS_MAINT_TUNNELS, ACCESS_HEADS_VAULT, ACCESS_MEDICAL, ACCESS_MIME,
 				ACCESS_MINING, ACCESS_MINING_STATION, ACCESS_MINERAL_STOREROOM, ACCESS_SMITH, ACCESS_MORGUE, ACCESS_NTREP, ACCESS_PARAMEDIC, ACCESS_ALL_PERSONAL_LOCKERS,
 				ACCESS_ENGINE_EQUIP, ACCESS_PSYCHIATRIST, ACCESS_QM, ACCESS_RD, ACCESS_RC_ANNOUNCE, ACCESS_ROBOTICS, ACCESS_TOX, ACCESS_RESEARCH, ACCESS_SECURITY,
-				ACCESS_SURGERY, ACCESS_TECH_STORAGE, ACCESS_TELEPORTER, ACCESS_THEATRE, ACCESS_TCOMSAT, ACCESS_TOX_STORAGE, ACCESS_VIROLOGY, ACCESS_WEAPONS, ACCESS_XENOBIOLOGY, ACCESS_TRAINER, ACCESS_EVIDENCE)
```
Don't know why this was changed to have a new line.

EmeraldCandy replied:
remnant of something else that didnt go through at the time. oops!

PollardTheDragon commented on code/game/objects/items/weapons/storage/surgical_tray.dm:
```diff
@@ -19,7 +19,8 @@
 		/obj/item/stack/medical/bruise_pack,
 		/obj/item/stack/medical/ointment,
-		/obj/item/surgical_drapes
+		/obj/item/surgical_drapes,
```
Instead of adding a dissector to every surgical tray, why not make a subtype for xenobiology that has the dissector? Or do we want the OR's in robotics and medbay to also have these?

1080pCat commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+	if(tgui_alert(src, "This is a permanent action, guarenteeing this person will be removed from the round. Are you sure?", "Insert Cursed Bananium Heart", list("Yes", "No")) != "Yes")
+		return
+	if(M.mind.assigned_role == "Clown")
+		addtimer(CALLBACK(src, PROC_REF(glorious_death), M), 5 MINUTES)
```
tg cooldown?

Qwertytoforty replied:
callback, not a cooldown

1080pCat commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+	else
+		M.emote("scream")
+		M.SetJitter(20 SECONDS)
+		addtimer(CALLBACK(M, TYPE_PROC_REF(/mob/living/carbon/human/, makeCluwne)), 5 SECONDS)
```
ditto

1080pCat commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+	M.Stun(10 SECONDS)
+	M.dir = 2
+	animate(M, pixel_y = 64, time = 2 SECONDS)
+	addtimer(CALLBACK(src, PROC_REF(finalize_death), M), 2 SECONDS)
```
ditto

1080pCat commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+		explosion(user.loc, 0, 2, 3, 3, cause = user)
+	if(quality == ORGAN_PRISTINE)
+		user.status_flags |= GODMODE
+		addtimer(CALLBACK(src, PROC_REF(cleanup), user), 0.3 SECONDS) // juuust long enough to survive the blast itself.
```
tg cooldown?

Qwertytoforty replied:
no this should be a timer

1080pCat commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+
+/obj/item/organ/internal/ears/xenobiology/colorful/on_life()
+	. = ..()
+	if(next_change > world.time)
+		return
+	next_change = world.time + 1 MINUTES
```
tg cooldown?

EmeraldCandy replied:
converted

1080pCat reviewed (commented):
woe, list standardization apon ye.
also tg cooldown stuff that im not *100%* sure if are appropriate? but im fairly sure
and general description stuff

EmeraldCandy commented on code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:
```diff
@@ -71,11 +70,41 @@ Difficulty: Hard
 	score_type = BUBBLEGUM_SCORE
 	deathmessage = "sinks into a pool of blood, fleeing the battle. You've won, for now... "
```
The idea was that he just kind of leaves a pile some some other entity's flesh. I dunno if bubblegum collects it or what, but I hinted at it in the description on death: _"Upon closer inspection it doesnt actually look like it comes from bubblegum, but some other kind of foreign entity."_

EmeraldCandy commented on code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm:
```diff
@@ -43,6 +43,7 @@ Difficulty: Hard
 	attack_sound = 'sound/weapons/sonic_jackhammer.ogg'
 	icon_state = "hierophant"
 	icon_living = "hierophant"
+	icon_dead ="hierophant_dead"
```
I imagined it as a core seperating from the staff. That big purple glowy bit in the hiero sprite isnt really on the staff so I saw it as that

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+	if(M.mind)
+		if(M.mind.assigned_role == "Clown")
+			M.AddComponent(/datum/component/squeak)
+			owner.move_speed = -0.2
```
it would but I should remove that, thanks. I was still tossing ideas around but put movement speed with the cursed heart only.

EmeraldCandy commented on code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:
```diff
@@ -55,12 +54,42 @@ Difficulty: Very Hard
 	deathmessage = "disintegrates, leaving a glowing core in its wake."
```
gave it a lil update

EmeraldCandy commented on code/modules/research/xenobiology/organ_analyzer.dm:
```diff
@@ -0,0 +1,192 @@
+		if(processing_organ)
+			to_chat(user, "<span class='warning'>The machine is already analyzing the organ!</span>")
+		else
+			time_to_complete = world.time + 15 SECONDS
```
done!

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+				owner.set_heartattack(TRUE) // yeah probably shouldnt use this
+		owner.AdjustConfused(5 SECONDS)
+		owner.vomit(20)
+		next_activation = world.time + 20 SECONDS
```
adjusted it, no cooldown needed

EmeraldCandy commented on code/modules/antagonists/changeling/powers/contort_body.dm:
```diff
@@ -26,9 +33,30 @@
+		duration = 1 MINUTES
+	next_activation = world.time + cooldown
+	addtimer(CALLBACK(src, PROC_REF(deactivate), user), duration)
+	. = ..()
```
updated

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+	. = ..()
+	var/atom/target = targets[1] //There is only ever one target
+	var/dist = get_dist(user.loc, target.loc)
+	if(target.z != user.z || dist > 2)
```
done!

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+		to_chat(user, "<span class = 'warning'>We are unable to teleport ourself!</span>")
+		revert_cast()
+		return
+	if(get_dist(target.loc, user.loc) > 2)
```
done!

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1514 @@
+	var/cooldown = 0
+	var/static/list/migo_sounds
+
+/datum/action/innate/migo_noise/New(Target)
```
I dont believe so since its linked to the mob itself, unless I move it into a global or something but then its just moving the list

lewcc replied:
This would be a great use of a global list, please do so

EmeraldCandy replied:
made global

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+/obj/item/organ/internal/liver/xenobiology/soupy
+	name = "Soupy Organ"
+	desc = "This organ seems to barely keep its own form together. It also reeks of tomato sauce."
+	analyzer_price = 15
+	var/original_name
+	var/original_own_species_blood
+	var/original_exotic_blood
+	var/original_blood_color
+	var/original_blood_type
+	hidden_origin_tech = TECH_BIO
+	hidden_tech_level = 6
```
New vars after inherited ones, please
```suggestion
	analyzer_price = 15
	hidden_origin_tech = TECH_BIO
	hidden_tech_level = 6
	var/original_name
	var/original_own_species_blood
	var/original_exotic_blood
	var/original_blood_color
	var/original_blood_type
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	M.dna.species.own_species_blood = TRUE
+	M.dna.species.exotic_blood = "tomatojuice"
+	M.dna.species.blood_color = "#e25821"
+	M.dna.blood_type = "Tomato"
```
this is terrifying. will people with this be able to get blood back by eating tomatoes, or are they SOL?

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+/datum/action/innate/migo_noise/Activate()
+	var/mob/living/carbon/human/user = owner
+	if(!COOLDOWN_FINISHED(src, migo_cooldown))
+		to_chat(user, "<span class='warning'>You cant get your noisy organ to speak again so soon!</span>")
```
```suggestion
		to_chat(user, "<span class='warning'>You can't get your noisy organ to speak again so soon!</span>")
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+
+/obj/item/organ/internal/eyes/xenobiology/receptors
+	name = "Photosensitive Receptors"
+	desc = "A set of organs receptive to light in the spectrum of- hey wait a second. Arn't these just eyes?"
```
```suggestion
	desc = "A set of organs receptive to light in the spectrum of -- hey, wait a second. Aren't these just eyes?"
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+		if(ORGAN_PRISTINE)
+			see_in_dark = 8
+			vision_flags = SEE_OBJS | SEE_TURFS // slightly better mesons
+	. = ..()
```
```suggestion
	return ..()
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	name = "Paradoxical Organ"
+	desc = "The organ is constantly shifting and morphing, each time you look away its something new."
+	analyzer_price = 40
+	var/list/acceptable_hearts = list()
```
Please autodoc what this variable means. Also, does it ever get set?

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+
+/obj/item/organ/internal/heart/xenobiology/paradox
+	name = "Paradoxical Organ"
+	desc = "The organ is constantly shifting and morphing, each time you look away its something new."
```
```suggestion
	desc = "The organ is constantly shifting and morphing; each time you look away it's something new."
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+/obj/item/organ/internal/heart/xenobiology/paradox/Initialize(mapload)
+	. = ..()
+	icon_state = pick("hiero1", "hiero2")
+/obj/item/organ/internal/heart/xenobiology/paradox/insert(mob/living/carbon/human/M, special, dont_remove_slot)
```
```suggestion

/obj/item/organ/internal/heart/xenobiology/paradox/insert(mob/living/carbon/human/M, special, dont_remove_slot)
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	)
+
+/obj/item/organ/internal/heart/xenobiology/cursed_bananium/insert(mob/living/carbon/human/M, special, dont_remove_slot)
+	if(tgui_alert(src, "This is a permanent action, guarenteeing this person will be removed from the round. Are you sure?", "Insert Cursed Bananium Heart", list("Yes", "No")) != "Yes")
```
```suggestion
	if(tgui_alert(src, "This is a permanent action, guaranteeing this person will be removed from the round. Are you sure?", "Insert Cursed Bananium Heart", list("Yes", "No")) != "Yes")
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	analyzer_price = 80
+	hidden_origin_tech = TECH_TOXINS
+	hidden_tech_level = 7
+/obj/item/organ/internal/cell/xenobiology/supercharged/Initialize(mapload)
```
```suggestion

/obj/item/organ/internal/cell/xenobiology/supercharged/Initialize(mapload)
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+
+/obj/item/organ/internal/heart/xenobiology/megacarp/insert(mob/living/carbon/human/M, special = 0, dont_remove_slot = 0)
+	if(!M.mind)
+		M.visible_message("The organ doesnt look like its going to fit right now! It refuses.")
```
```suggestion
		M.visible_message("[src] doesn't look like it's going to fit right now! It...refuses?")
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	var/datum/spell/shapeshift/megacarp/spell = new
+	spell.quality = organ_quality
+	M.mind.AddSpell(spell)
+	. = ..()
```
Please convert these trailing `. = ..()`s to `return ..()`, I'm going to leave those up to you to catch for the rest of this file

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+		possible_shapes = list(/mob/living/simple_animal/hostile/carp/xeno_organ)
+
+/datum/spell/shapeshift/megacarp/Shapeshift(mob/living/carbon/human/M)
+	M.visible_message("<span class='danger'>[M] screams in agony as scales and fins erupt out of their flesh!</span>",
```
```suggestion
	M.visible_message("<span class='danger'>[M] screams in agony as scales and fins erupt out of [M.p_their()] flesh!</span>",
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+
+/datum/spell/shapeshift/megacarp/Shapeshift(mob/living/carbon/human/M)
+	M.visible_message("<span class='danger'>[M] screams in agony as scales and fins erupt out of their flesh!</span>",
+		"<span class='dangeruser'>You begin channeling the painful transformation.</span>")
```
```suggestion
		"<span class='userdanger'>You begin channeling the painful transformation.</span>")
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+		if(ORGAN_PRISTINE)
+			owner.bodytemperature -= 30
+
+/obj/item/organ/internal/kidneys/xenobiology/lethargic
```
honestly this would be great loot for the sloth ruin

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+		revert_cast()
+		return
+	if(target == user)
+		to_chat(user, "<span class='warning'>It would be a bad idea to target ourself.</span>")
```
```suggestion
		to_chat(user, "<span class='warning'>It would be a bad idea to target ourselves.</span>")
```

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+		if(quality == ORGAN_DAMAGED && prob(30))
+			user.adjustBruteLoss(15) // ouch!
+		user.visible_message("<span class='warning'>[src] produces a new head!</span>", "<span class='warning'>You produce a new head and send it to your target!</span>")
+		newhead = new /mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/xenobiology(user.loc)
+		newhead.GiveTarget(target)
+		newhead.faction = user.faction.Copy()
+		newhead.icon = null
+		newhead.name = "[user]'s head"
+		newhead.parent_spell = src
+		newhead.faction = list(user)
+		newhead.permanent_target = target
+		var/obj/item/user_head = user.get_organ("head")
+		var/obj/item/dupe_head = DuplicateObject(user_head, perfectcopy=1, sameloc=0, newloc=newhead.contents)
+		newhead.mounted_head = dupe_head
+		var/matrix/M = matrix()
+		dupe_head.transform = M
+		var/image/IM = image(dupe_head.icon, dupe_head.icon_state)
+		IM.pixel_y -= 6
+		IM.overlays = dupe_head.overlays.Copy()
+		newhead.overlays += IM
```
Could you move all of this logic into the Initialize() of the `.../legionnairehead/xenobiology` from the user and target?

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/xenobiology/Life(seconds, times_fired)
+	if(target != permanent_target)
+		src.visible_message("<span class ='notice'>With no valid targets, the head crumbles into a pile of flesh</span>")
+		parent_spell.on_head_death()
```
Can you register a signal instead of doing it like this?

lewcc commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	density = FALSE
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	var/datum/spell/head_attack/parent_spell
+	var/mob/living/permanent_target
+	var/obj/item/organ/external/head/mounted_head
```
Make sure these are nulled out on Destroy() please

lewcc commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -9,6 +9,19 @@
+	var/organ_quality = ORGAN_NORMAL
+	/// Does this organ originate from the xenobiology dissection loop?
+	var/is_xeno_organ = FALSE
+	var/cooldown_low = 300
+	var/cooldown_high = 300
+	var/next_activation = 0
 
```
Please document what these cooldown variables are in reference to (it's not very clear at a glance that they're related to xeno organs), maybe name it something like `organ_ability_cooldown` or `organ_action_cooldown`?

lewcc commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -34,6 +47,11 @@
+/obj/item/organ/internal/examine(mob/user)
+	. = ..()
+	if(is_xeno_organ)
+		. += "<span class='info'>It looks like it would replace the [slot]."
```
```suggestion
		. += "<span class='info'>It looks like it would replace \the [slot]."
```

lewcc commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -216,6 +234,22 @@
+			qdel(used)
+			return ITEM_INTERACT_COMPLETE
+		else
+			to_chat(user, "<span class='warnin'>This organ is already pristine!</span>")
```
```suggestion
			to_chat(user, "<span class='warning'>This organ is already in pristine condition!</span>")
```

lewcc commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -216,6 +234,22 @@
+		if(organ_quality < ORGAN_PRISTINE)
+			organ_quality = ORGAN_PRISTINE
+			to_chat(user, "<span class='info'>You apply the regenerative mesh to the organ. It now looks pristine!</span>")
+			name = "Pristine [name]"
```
```suggestion
			name = "pristine [name]"
```

lewcc commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -394,3 +428,11 @@
 	REMOVE_TRAIT(src, TRAIT_ORGAN_INSERTED_WHILE_DEAD, "[UID()]")
 	UnregisterSignal(owner, COMSIG_LIVING_DEFIBBED)
+
+/obj/item/organ/internal/proc/Start()
+	return
```
Is this used by anything? If so, please change the name to `start()` and add a comment explaining what it's supposed to be used for.

lewcc reviewed (changes requested):
Time for a review bomb. I haven't really been paying too much attention to this feature, but it looks really cool. I love in-game experiments like this where you never know what you're gonna get.

One thing I kind of want to bookend this review with is that I think I'd like to give the surgery stuff a second look, I'm not entirely sure if I agree with how it's implemented with list pairs and arbitrary surgery steps and messages. I honestly think it's a bit fragile, and we could probably work on a better, more modular approach. Feel free to poke me down the line about this.

EmeraldCandy commented on code/__DEFINES/mob_defines.dm:
```diff
@@ -16,6 +16,11 @@
 #define ORGAN_DATUM_BATTERY "battery"
 
+// Organ quality datums.
+#define ORGAN_DAMAGED 1
```
its less of a check for "if its broken" and moreso a statement of their qualities, of 3 tiers. Damaged, normal, and pristine. These are not equated to how much damage they've taken, but how powerful the organs effects are and is decided upon organ extraction. An organ can be damaged, but at full health. likewise, an organ can be "pristine" but broken and non-functional as an organ.

EmeraldCandy commented on code/game/objects/effects/spawners/random/traders/trader_department_spawners.dm:
```diff
@@ -94,6 +94,8 @@
 		// Xenobio
 		/obj/item/slimepotion/sentience = 50, // Low-value, but we want to encourage getting more players back in the round.
 		/obj/item/slimepotion/transference = 50,
+		/obj/item/dissector/alien = 50,
+		/obj/item/regen_mesh = 50,
```
done

EmeraldCandy commented on code/modules/antagonists/changeling/powers/contort_body.dm:
```diff
@@ -6,6 +6,13 @@
 	dna_cost = 4
 	power_type = CHANGELING_PURCHASABLE_POWER
 	category = /datum/changeling_power_category/utility
+	var/organ_quality = ORGAN_NORMAL
```
i mostly didnt want to dupe code where I could since they're more or less the same thing

EmeraldCandy replied:
converting into its own power via some code dupe

EmeraldCandy commented on code/modules/mob/living/living.dm:
```diff
@@ -907,6 +908,77 @@
+		return ORGAN_NORMAL
+	return ORGAN_DAMAGED
+
+/mob/living/examine(mob/user)
```
there is no other `/mob/living/examine()` that i was able to find even with the search across all files

EmeraldCandy commented on code/modules/mob/living/simple_animal/hostile/megafauna/ancient_robot.dm:
```diff
@@ -146,6 +172,14 @@ Difficulty: Hard
 
 /mob/living/simple_animal/hostile/megafauna/ancient_robot/death(gibbed, allowed = FALSE)
 	if(allowed)
+		overlays.Cut()
+		underlays.Cut()
+		QDEL_NULL(TR)
+		QDEL_NULL(TL)
+		QDEL_NULL(BR)
+		QDEL_NULL(BL)
+		QDEL_NULL(beam)
+		body_shield_enabled = FALSE
```
the corpses being left over, you mean? this change on death was just to make sure the legs and overlays and the like weren't left over after it explodes and dies.

EmeraldCandy commented on code/modules/mob/living/simple_animal/slime/slime_life.dm:
```diff
@@ -476,3 +479,23 @@
 		return TRUE
 	return TRUE
+
+/mob/living/simple_animal/slime/proc/handle_organs()
```
done

EmeraldCandy commented on code/modules/mob/living/simple_animal/slime/slime_life.dm:
```diff
@@ -476,3 +479,23 @@
+	finished_organ.organ_quality = holding_organ.unknown_quality
+	finished_organ.icon_state = holding_organ.icon_state
+	if(finished_organ.organ_quality != ORGAN_NORMAL)
+		finished_organ.name = "[finished_organ.organ_quality == ORGAN_PRISTINE ? "Pristine" : finished_organ.organ_quality == ORGAN_NORMAL ? "Standard" : "Damaged"] [finished_organ.name]"
```
made!

EmeraldCandy commented on code/modules/mob/living/simple_animal/slime/slime_mob.dm:
```diff
@@ -111,6 +119,20 @@
+/mob/living/simple_animal/slime/update_overlays()
+	. = ..()
+	if(holding_organ && stat != DEAD)
+		var/mutable_appearance/underlay_appearance = mutable_appearance(layer = 3.9, plane = GAME_PLANE)
```
added a note after discord talk

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	origin_tech = null
+	tough = TRUE
+	is_xeno_organ = TRUE
+	var/can_paradox = FALSE
```
documented

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+		if(ORGAN_PRISTINE) // careful, it removes good shit too
+			owner.adjustToxLoss(-3)
+			for(var/datum/reagent/R in owner.reagents.reagent_list)
+				if(R != src)
```
my bad, accidently left that in

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	if(!isunathi(M))
+		for(var/datum/action/innate/unathi_ignite/fire in M.actions)
+			fire.Remove(M)
+	M.RemoveSpell(/datum/spell/drake_breath)
```
discussed on discord, wont be an issue

EmeraldCandy commented on code/modules/antagonists/changeling/powers/contort_body.dm:
```diff
@@ -26,9 +33,30 @@
+		duration = 1 MINUTES
+	COOLDOWN_START(src, organ_cooldown, cooldown)
+	addtimer(CALLBACK(src, PROC_REF(deactivate), user), duration)
+	. = ..()
```
removed section

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1533 @@
+	action_background_icon_state = "bg_default"
+	sound = null
+	active = FALSE
+	var/quality
```
yes. if its pristine it gets passed down, and will do 10 brute damage to someone when hit

1080pCat reviewed (changes requested):
Few issues i've found.

- The IMS acts as a dissector, unsure if this is intended as its the same speed as the abductor one.

- The dissector/IMS "baps" the target after you've done the organ extraction step
<details>
-<summary>The IMS runtimes on a successful organ extraction</summary>
<br>

```dm
[2025-05-05T14:55:47] Runtime in code/modules/mob/living/living.dm:947: pick() from empty list
   proc name: attempt dissection (/mob/living/proc/attempt_dissection)
   usr: Alex Campbell (1080pcat) (/mob/living/carbon/human)
   usr.loc: The shuttle floor (130,127,2) (/turf/simulated/floor/mineral/titanium/blue)
   src: the goliath (/mob/living/simple_animal/hostile/asteroid/goliath/beast)
   src.loc: the shuttle floor (129,126,2) (/turf/simulated/floor/mineral/titanium/blue)
   call stack:
   the goliath (/mob/living/simple_animal/hostile/asteroid/goliath/beast): attempt_dissection
   the goliath (/mob/living/simple_animal/hostile/asteroid/goliath/beast): attack_by
   the incision management system (/obj/item/scalpel/laser/manager): melee_attack_chain
   Alex Campbell (/mob/living/carbon/human): ClickOn
   the goliath (/mob/living/simple_animal/hostile/asteroid/goliath/beast): Click
```

</details>


- The saw step doesn't play the correct sound, and only plays the sound once you're done, not when you start(inconsistent with the other tools)
-
- the "You struggle to perform the dissection properly, and will have to start the last step over!" prompt doesn't do anything, you can just freely continue 

- bubbles, colossus, and the legions bodies are all weirdly offset
![image](https://github.com/user-attachments/assets/ed209fc9-00b7-457f-8fdb-221760ff3c03)

<details>
-<summary>Vetus throws 5 separate runtimes on death, relating to its legs</summary>
<br>

```dm
[2025-05-05T15:08:34] Runtime in code/modules/mob/living/simple_animal/hostile/megafauna/ancient_robot.dm:651: Cannot execute null.forceMove().
   proc name: Moved (/mob/living/simple_animal/hostile/megafauna/ancient_robot/Moved)

[2025-05-05T15:08:34] Runtime in code/modules/mob/living/simple_animal/hostile/megafauna/ancient_robot.dm:601/603/605/607: Cannot execute null.leg movement().
   proc name: leg control system (/mob/living/simple_animal/hostile/megafauna/ancient_robot/proc/leg_control_system)
```

</details>
- the space dragon counts as an ash drake (because its a subtype of it for whatever reason

- lesser ash drakes(the one you can turn into via dragons blood) also suffers from this issue

- hardmode bubbles drops two bodies, one on first "death" second on second real..er death(in its arena) Probably not the *biggest* deal since its left behind, but a miner with tools could probably get a second organ from the arena body.

- Related, bubbles will pull the first body to the arena with you.

- Doesn't seem like you can dissect the fleshling?

EmeraldCandy commented:
> * The IMS acts as a dissector, unsure if this is intended as its the same speed as the abductor one.
IMS can be used as a dissector.  Though this isnt a great idea because of high chances of organ quality being undesirable. scalpels, wirecutters, and even forks can be used (but you're nearly gaurenteed a damaged organ)
> * The dissector/IMS "baps" the target after you've done the organ extraction step

> The IMS runtimes on a successful organ extraction
wasnt related tot he IMS, but still fixed.

> * The saw step doesn't play the correct sound, and only plays the sound once you're done, not when you start(inconsistent with the other tools)
im... going to look soon into making it so this is an actual surgery and not some snowflakey mess
> * bubbles, colossus, and the legions bodies are all weirdly offset
moved images into a new 32x32 file and changed icon on death to use the new file.
> Vetus throws 5 separate runtimes on death, relating to its legs
I couldnt get this to proc myself, however I tried to add some checks because I believe its related to moving and the leg controller subsystem.
> - the space dragon counts as an ash drake (because its a subtype of it for whatever reason)
space dragon is intended. Lesser is not. Removed xeno organs from lessers.
> * hardmode bubbles drops two bodies, one on first "death" second on second real..er death(in its arena) Probably not the _biggest_ deal since its left behind, but a miner with tools could probably get a second organ from the arena body.
ensures that hardmode bubble doesnt drop an initial corpse, and that it wont drag dead bodies in with it to its arena.
> * Doesn't seem like you can dissect the fleshling?
fleshling's item isnt an organ. They'll need to be butchered for it.

SpecialX989 commented:
Quick question on this, I see it says megafauna are not pullable, does that mean that you can no longer transport with a locker or bodybag like you could with drake corpses, or are they truly immovable now.

Qwertytoforty commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1664 @@
+	var/atom/target = targets[1]
+	if(!isliving(target))
+		to_chat(user, "<span class = 'warning'>We can only teleport living things!</span>")
+		return
```
have it revert_cast() here

Qwertytoforty commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1664 @@
+	. = ..()
+	M.RemoveSpell(/datum/spell/turf_teleport/organ_teleport)
+
+/datum/spell/turf_teleport/organ_teleport
```
this spell should probably adjust 
	var/inner_tele_radius = 1
	var/outer_tele_radius = 2
	to something like 10 to 20 or it just moves them like 2 tiles

EmeraldCandy replied:
adjusted

Baximusu commented:
Was asked to comment about it, when you kill the Hierophant, as there is still a 'body' left behind, the music continues to play to the end.

Qwertytoforty commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,1664 @@
+/obj/item/organ/internal/heart/xenobiology/paradox/insert(mob/living/carbon/human/M, special, dont_remove_slot)
+	for(var/obj/item/organ/internal/heart/xenobiology/temp_organ as anything in subtypesof(/obj/item/organ/internal/heart/xenobiology))
+		if(temp_organ::can_paradox)
+			acceptable_hearts += temp_organ
```
May I suggest excluding the "hearts that kill you" hearts, considering rareity

EmeraldCandy replied:
fixed

Qwertytoforty commented:
disection manager should probably fit in the surgical belt

EmeraldCandy commented:
changelog:

Fixed organ quality
fixed hound/migo organs being swapped
added dissector to possible surgical belt storage options
removed vestigial heart from pandora mix
added missing revert_cast on heiro organ when targeting nonliving things
adjusted heiro organ teleport distance calculation
gave xenobiologists teleporter access
fixed some megafauna becomming invisible after death
megafauna can no longer be put into lockers

EmeraldCandy commented on code/modules/surgery/organs_internal.dm:
```diff
@@ -401,6 +401,9 @@
 			chat_message_type = MESSAGE_TYPE_COMBAT
 		)
+	if(I.warning)
+		if(tgui_alert(user, "This is a permanent action, guaranteeing this person will be removed from the round. Are you sure?", "Insert Cursed Bananium Heart", list("Yes", "No")) != "Yes")
+			return
```
added

EmeraldCandy commented:
Organ extractors can now be used in place of a dissector too. When used as the final step, you are granted the organ without needing slime revitalization. Organ Quality chance is better than base extractor but worse than upgraded when used

warriorstar-orion commented on code/modules/mob/living/simple_animal/slime/slime_mob.dm:
```diff
@@ -64,6 +66,12 @@
+	// does this slime have a xeno organ inserted into it already?
+	var/obj/item/xeno_organ/holding_organ
+	// An elevated form of discipline. Is this slime trained and ready to begin organ therapy?
+	var/trained = 0
```
```suggestion
	var/trained = FALSE
```

warriorstar-orion commented on code/modules/mob/living/simple_animal/slime/slime_mob.dm:
```diff
@@ -355,18 +385,40 @@
+		if(S.amount < 5)
+			to_chat(user, "<span class='notice'>You need at least five sheets of plasma to feed the slime!</span>")
+			return ITEM_INTERACT_COMPLETE
+		new /obj/effect/temp_visual/heart(src.loc)
+		src.visible_message("<span class='notice'>[user] feeds the slime some plasma. It chirps happily!</span>", "<span class='notice'>[user] feeds you a few sheets of plasma! Yummy!!!</span>")
```
```suggestion
		new /obj/effect/temp_visual/heart(loc)
		visible_message("<span class='notice'>[user] feeds the slime some plasma. It chirps happily!</span>", "<span class='notice'>[user] feeds you a few sheets of plasma! Yummy!!!</span>")
```

EmeraldCandy commented on code/__DEFINES/job_defines.dm:
```diff
@@ -57,3 +57,4 @@
 #define JOB_ASSISTANT			(1<<13)
 #define JOB_EXPLORER			(1<<14)
 #define JOB_SMITH				(1<<15)
+#define JOB_XENOBIOLOGIST		(1<<15)
```
that was on me, i big goofed when putting that in. I probably meant to continue it and got distracted

warriorstar-orion commented on code/modules/mob/living/living.dm:
```diff
@@ -232,7 +232,7 @@
 	if(dissection)
 		. += "<span class='notice'>You detect the next dissection step will be: [dissection.get_surgery_step()]</span>"
-	if(!xeno_organ_results)
+	if(!surgery_container.xeno_generic_organs)
```
should this also check the contents of xeno_specialized_organs?

EmeraldCandy replied:
no because the specialized_organs isnt really touched, but generic organs are nulled. This is just a way to tell if they've had surgery done on them. I cant use contains_xeno_organ because most things dont have it.

BAGELMENSK commented:
Don't know if this has been reported, but I had a round where I couldnt do the dissection surgery. I was able to do it in the round previous, but in the next round it jsut gave me the "There are no surgeries you can perform on this right now"(not the exact message, quoting from memory, was blue text) message.

BAGELMENSK commented:
Also in that same round, when I went to do normal surgery on Spats, There were 2 organ manipulation surgeris in the lower torso section?????

EmeraldCandy commented:
> Also in that same round, when I went to do normal surgery on Spats, There were 2 organ manipulation surgeris in the lower torso section?????

found and fixed

EmeraldCandy commented:
> Don't know if this has been reported, but I had a round where I couldnt do the dissection surgery. I was able to do it in the round previous, but in the next round it jsut gave me the "There are no surgeries you can perform on this right now"(not the exact message, quoting from memory, was blue text) message.

hard to pinpoint that without being in round to diagnose, but make sure you're targeting the chest

BAGELMENSK commented:
On round 44854 right now, the dissection surgery wont give me the organ after minutes of doign the final step over and over on three watcher bodies. OTher Xenobiologist says that the Basilisk didnt require the usual basilisk surgery to extract from, but it did work. NEither of us are gettign surgery step prom;pts (Step fail/success red/blue text) only stopping the surgery gives prompt text.

warriorstar-orion commented on code/datums/components/surgery_initiator.dm:
```diff
@@ -314,21 +314,10 @@
-			procedure.steps = target.surgery_container.dissection_tool_step
+			log_debug("A dissection was started on [target] with contains_xeno_organ as TRUE, however its surgery_container was null!")
+			to_chat(user, "<span class = 'userdanger'>[target] does not have a dissection surgery information set to it. Please inform an admin or developer.</span>")
+		target.surgery_container = new target.surgery_container
```
```suggestion
		if(ispath(target.surgery_container, /datum/xenobiology_surgery_container))
			target.surgery_container = new target.surgery_container
```

EmeraldCandy commented on code/modules/mob/living/living.dm:
```diff
@@ -1237,3 +1252,12 @@
+			for(var/datum/surgery/S in surgeries)
+				if(S.next_step(user, src))
+					return ITEM_INTERACT_COMPLETE
+
```
it shouldnt, it was in carbon before and I just moved it up one parent

EmeraldCandy commented on code/modules/research/xenobiology/xenobiology_dissections.dm:
```diff
@@ -0,0 +1,838 @@
+	custom_organ_states = list("colossus1", "colossus2", "colossus3", "colossus4")
+
+	dissection_tool_step = list(
+	/datum/surgery_step/generic/dissect,
+	/datum/surgery_step/generic/cut_open,
+	/datum/surgery_step/generic/clamp_bleeders,
+	/datum/surgery_step/generic/retract_skin,
+	/datum/surgery_step/generic/cut_open,
+	/datum/surgery_step/generic/dissect,
```
done

EmeraldCandy commented on code/modules/research/xenobiology/Xenobiology_organs.dm:
```diff
@@ -0,0 +1,1576 @@
+			fire.Remove(M)
+	M.RemoveSpell(/datum/spell/drake_breath)
+
+/datum/spell/drake_breath
+	name = "Drake Breath"
+	desc = "Draw from special glands in your body to produce a small, but dazzling flame!"
+	base_cooldown = 3 MINUTES
+	clothes_req = FALSE
+	stat_allowed = CONSCIOUS
+	invocation_type = "none"
+	action_icon_state = "fireball0"
```
applied!

Burzah reviewed (approved):
Given the length of the TM and the changes made with the suggestions other reviewers, I say it is ready to go.

Approved.

Burzah commented on code/modules/mining/lavaland/loot/hierophant_loot.dm:
```diff
@@ -224,6 +224,8 @@
 			var/obj/effect/temp_visual/hierophant/blast/B = new /obj/effect/temp_visual/hierophant/blast(t, user, TRUE) //but absolutely will hurt enemies
 			B.damage = 30
 		for(var/mob/living/L in range(1, source))
+			if(ismegafauna(L) %% L.stat == DEAD)
```
```suggestion
			if(ismegafauna(L) && L.stat == DEAD)
```
Causing a runtime (division by zero). Modulo moment.

lewcc reviewed (commented):
Went through again, mostly grammar and little nits!

warriorstar-orion commented on code/modules/mob/living/simple_animal/slime/slime_life.dm:
```diff
@@ -476,3 +479,26 @@
+	finished_organ.icon_state = holding_organ.icon_state
+	finished_organ.name = "[quality_to_string(finished_organ.organ_quality, FALSE)] [finished_organ.name]"
+	underlays.Cut()
+	qdel(holding_organ)
+	holding_organ = null
```
QDEL_NULL is defined as `#define QDEL_NULL(item) if(item) { qdel(item); item = null }` which makes me think the original order is correct (or that this should just be `QDEL_NULL(holding_organ)` anyway)

EmeraldCandy replied:
fixed with a qdel_null

EmeraldCandy commented on code/game/objects/structures/mirror.dm:
```diff
@@ -53,6 +55,11 @@
 
+/obj/structure/mirror/organ/obj_break(damage_flag, mapload)
+	playsound(src, "shatter", 70, TRUE)
+	GLOB.mirrors -= src
```
removed

EmeraldCandy commented on code/modules/mob/living/simple_animal/hostile/angel_statue.dm:
```diff
@@ -167,6 +167,7 @@
 	base_cooldown = 300
 	clothes_req = FALSE
 	aoe_range = 14
+	var/from_organ = FALSE
```
doc'd

EmeraldCandy commented on code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm:
```diff
@@ -101,6 +105,13 @@ Difficulty: Medium
 	invisibility = 100
 
+/mob/living/simple_animal/hostile/megafauna/dragon/death(gibbed)
+	move_force = MOVE_FORCE_OVERPOWERING
+	move_resist = MOVE_FORCE_OVERPOWERING
+	pull_force = MOVE_FORCE_OVERPOWERING
+	mob_size = MOB_SIZE_LARGE
```
added to CL

EmeraldCandy commented on code/modules/research/xenobiology/Xenobiology_organs.dm:
made lowercase

EmeraldCandy commented on code/modules/research/xenobiology/Xenobiology_organs.dm:
```diff
@@ -0,0 +1,1606 @@
+
+// We dont want to keep the unidentified organ as an implantable version to skip the research phase
+/obj/item/xeno_organ
+	name = "Unidentified Mass"
```
made lowercase

EmeraldCandy commented:
because batch merge is funky and now the suggestions are outdated I did them all manually

warriorstar-orion reviewed (approved):
TMed for nearly 200 rounds, code LGTM, all approvals received, sending.

lewcc reviewed (approved):
Been through a good few rounds of review, thanks so much for sticking with it and embarking on structural changes when asked. 

LGTM.
